sihl_users/lib/repository_mariadb.ml

```diff
@@ -0,0 +1,308 @@
+module Sql = struct
+  let set_fk_check connection status =
+    let module Connection = (val connection : Caqti_lwt.CONNECTION) in
+    let request =
+      Caqti_request.exec Caqti_type.bool
+        {sql|
+        SET FOREIGN_KEY_CHECKS = ?;
+           |sql}
+    in
+    Connection.exec request status
+
+  module User = struct
+    module Model = struct
+      open Model.User
+
+      let t =
+        let encode m =
+          Ok
+            ( m.id,
+              ( m.email,
+                ( m.username,
+                  ( m.password,
+                    (m.name, (m.phone, (m.status, (m.admin, m.confirmed)))) ) )
+              ) )
+        in
+        let decode
+            ( id,
+              ( email,
+                ( username,
+                  (password, (name, (phone, (status, (admin, confirmed))))) ) )
+            ) =
+          let _ = Logs.info (fun m -> m "received user with id %s" id) in
+          Ok
+            {
+              id;
+              email;
+              username;
+              password;
+              name;
+              phone;
+              status;
+              admin;
+              confirmed;
+            }
+        in
+        Caqti_type.(
+          custom ~encode ~decode
+            (tup2 string
+               (tup2 string
+                  (tup2 string
+                     (tup2 string
+                        (tup2 string
+                           (tup2 (option string) (tup2 string (tup2 bool bool)))))))))
+    end
+
+    let get_all connection =
+      let module Connection = (val connection : Caqti_lwt.CONNECTION) in
+      let request =
+        Caqti_request.find Caqti_type.unit Model.t
+          {sql|
+        SELECT
+          LOWER(CONCAT(
+           SUBSTR(HEX(uuid), 1, 8), '-',
```

Comment:
Is this big/small endian safe? I'm not sure so we need to add item in our backlog to make sure.

Reply:
Correct me if I am wrong, but isn't this not a problem as long as `BINARY` is always little endian?

Reply:
That's the problem, I think this is not guaranteed. Anyway, very minor thing and just pop into my mind when looking at the code.